contracts/root/ValidatorSet.sol

```diff
@@ -0,0 +1,72 @@
+pragma solidity ^0.4.24;
+
+import { SafeMath } from "openzeppelin-solidity/contracts/math/SafeMath.sol";
+
+import { RootChainable } from "../mixin/RootChainable.sol";
+import { Lockable } from "../mixin/Lockable.sol";
+
+
+contract ValidatorSet is RootChainable, Lockable {
+  using SafeMath for uint256;
+  using SafeMath for uint8;
+  int256 constant INT256_MIN = -int256((2**255)-1);
+  uint256 constant UINT256_MAX = (2**256)-1; 
+  
+  event NewProposer(address indexed user, bytes data); 
+ 
+  struct Validator {
+    uint256 votingPower;
+    int256 accumulator;
+    address validator;
+  } // TODO: can use validator data from stakemanager.
+
+  address public proposer;
+  uint256 public totalVotingPower;
+  uint256 public lowestPower; 
+  // uint256 public totalEpochs;
+  Validator[] public validators;
+  
+
+  constructor() public {
+    proposer = address(0);
+    totalVotingPower = 0; 
+    lowestPower =  UINT256_MAX;
+  }
+  
+  function addValidator(address validator, uint256 votingPower) public {
+    require(votingPower > 0);
+    validators.push(Validator(votingPower, 0, validator)); //use index instead
+    if(lowestPower > votingPower ) { 
+      lowestPower = votingPower;
+    }
+    totalVotingPower += votingPower;
+  }
+
+  function _getProposer() public returns(address) {
```

Comment:
Let's not use `_` when function is public.

Reply:
can we change the name if it is changing the state? `changeProposer`/`selectProposer` or something?

Reply:
changed it to selectProposer, and merged both functions to save on gas.